src/main/java/org/wildfly/security/http/impl/SpnegoAuthenticationMechanism.java

```diff
@@ -99,71 +100,78 @@ public void evaluateRequest(HttpServerRequest request) throws HttpAuthentication
             }
 
             if (gssCredential == null) {
-                // We can't obtain a credential, we have no chance of performing SPNEGO authentication so give up now.
-                log.trace("Obtaining GSSCredential from callbackHandler failed - null credential");
+                log.trace("GSSCredential from callbackHandler is null - cannot perform SPNEGO authentication");
                 request.noAuthenticationInProgress();
                 return;
             }
 
-            List<String> authorizationValues = request.getRequestHeaderValues(AUTHORIZATION);
-            Optional<String> challenge = authorizationValues != null ? authorizationValues.stream()
-                    .filter(s -> s.startsWith(CHALLENGE_PREFIX)).limit(1).map(s -> s.substring(CHALLENGE_PREFIX.length()))
-                    .findFirst() : Optional.empty();
-
-            if (log.isTraceEnabled()) {
-                log.tracef("Sent HTTP authorizations: [%s]", authorizationValues == null ? "null" : String.join(", ", authorizationValues));
-            }
+            try {
+                gssContext = GSSManager.getInstance().createContext(gssCredential);
```

Comment:
Maybe this isn't the right time to notice this, but this mechanism (and its factory) should probably accept a GSSManager parameter whose default value is GSSManager.getInstance().  See org.wildfly.security.sasl.gs2.Gs2SaslServerFactory for reference.